### ISSUE TITLE
Bug: CS0419 if two assemblies contain the same type, but the type is internal to one assembly

### ISSUE THREAD

**Issue by qmfrederik** [dotnet/roslyn#10694]
**Version Used**: Visual Studio 2015 Update 2

**Steps to Reproduce**:

1. Create a new C# project
2. Enable XML documentation
3. Reference the following NuGet packages: `BouncyCastle.Crypto` and `Portable.Licensing`
4. On your main class, add the following method
```
using Org.BouncyCastle.Crypto.Parameters;
using System.Security.Cryptography;

(...)

        /// <summary>
        /// Converts a <see cref="RsaPrivateCrtKeyParameters"/> private key to a standard .NET
        /// private key.
        /// </summary>
        private static RSA ToRSA(RsaPrivateCrtKeyParameters privKey)
        {
            throw new NotImplementedException();
        }
```
5. Compile

Also see attached
[ReferenceBug.zip](https://github.com/dotnet/roslyn/files/226719/ReferenceBug.zip) project.

**Expected Behavior**:
The `cref` does not generate a warning:
* `BouncyCastle.Crypto` contains a public `RsaPrivateCrtKeyParameters` class; whereas `Portable.Licensing` contains an internal`RsaPrivateCrtKeyParameters` class, the reference is unambigious (as also demonstrated by the fact that the code compiles correctly).

**Actual Behavior**:
The following warning is generated:

```
Warning	CS0419	Ambiguous reference in cref attribute: 'RsaPrivateCrtKeyParameters'. Assuming 'RsaPrivateCrtKeyParameters', but could have also matched other overloads including 'RsaPrivateCrtKeyParameters'.
```

**Comment by 0xd4d** [comment]
See #77

You can probably use the full name of the type to remove the warning.

**Comment by gafter** [comment]
Dup of #4050 #4033
